src/ompl/geometric/planners/rrt/src/STRRTstar.cpp

```diff
@@ -0,0 +1,1114 @@
+                        tgi.xstate = si_->allocState();
+                        tgi.start = false;
+                        std::vector<Motion *> nbh;
+                        GrowState gsc = growTree(tGoal_, tgi, queue.front(), nbh, true);
+                        // connection successful, add all descendants and check if a new solution was found.
+                        if (gsc == REACHED)
+                        {
+                            // the motion was copied and added to the tree with a new parent
+                            // adjust children and parent pointers
+                            tgi.xmotion->children = queue.front()->children;
+                            for (auto &c : tgi.xmotion->children)
+                            {
+                                c->parent = tgi.xmotion;
+                            }
+                            tgi.xmotion->connectionPoint = queue.front()->connectionPoint;
+                            tgi.xmotion->numConnections = queue.front()->numConnections;
+                            Motion *p = tgi.xmotion->parent;
+                            while (p != nullptr)
+                            {
+                                p->numConnections += tgi.xmotion->numConnections;
+                                p = p->parent;
+                            }
+                            addDescendants(tgi.xmotion, tGoal_);
+                            // new solution found
+                            if (tgi.xmotion->numConnections > 0 &&
+                                tgi.xmotion->root->as<ompl::base::CompoundState>()
+                                        ->as<ompl::base::TimeStateSpace::StateType>(1)
+                                        ->position < bestSolutionTime)
+                            {
+                                bestSolutionTime = tgi.xmotion->root->as<ompl::base::CompoundState>()
+                                                       ->as<ompl::base::TimeStateSpace::StateType>(1)
+                                                       ->position;
+                                std::queue<Motion *> connectionQueue;
+                                connectionQueue.push(tgi.xmotion);
+                                while (!connectionQueue.empty())
+                                {
+                                    if (connectionQueue.front()->connectionPoint != nullptr)
+                                    {
+                                        solutionMotion = connectionQueue.front();
+                                        break;
+                                    }
+                                    else
+                                    {
+                                        for (Motion *c : connectionQueue.front()->children)
+                                            connectionQueue.push(c);
+                                    }
+                                    connectionQueue.pop();
+                                }
```

Comment:
You might also want to put this into a new function like "ComputeSolutionMotion" or something.